packages/dd-trace/src/lambda/handler.js

```diff
@@ -83,21 +83,9 @@ function extractContext (args) {
  */
 exports.datadog = function datadog (lambdaHandler) {
   return (...args) => {
-    const patched = lambdaHandler.apply(this, args)
+    const context = extractContext(args)
 
-    try {
-      const context = extractContext(args)
-
-      checkTimeout(context)
-
-      if (patched) {
-        // clear the timeout as soon as a result is returned
-        patched.then(_ => clearTimeout(__lambdaTimeout))
-      }
-    } catch (e) {
-      log.debug('Error patching AWS Lambda handler. Timeout spans will not be generated.')
-    }
-
-    return patched
+    checkTimeout(context)
+    return lambdaHandler.apply(this, args).then((res) => { clearTimeout(__lambdaTimeout); return res })
```

Comment:
shouldn't we catch in case an error gets thrown from our side? for debug purposes?

Reply:
`apply` in this case calls the handler method, so error handling is already performed by the lambda runtime. Anything we'd do would just log and re-throw the error anyway, so I'm not sure we need anything additional here.